scripts/gallery/team.py

```diff
@@ -0,0 +1,261 @@
+
+	while components:
+		try:
+			#obj = importlib.import_module(".".join(components))
+			obj = importlib.import_module(name)
+		except ImportError:
+			attrs.insert(0, components(pop))
+		else:
+			break
+	if not obj:
+		raise ImportError(f'Could not import "{obj_as_str}"')
+
+	for attr in attrs:
+		obj = getattr(obj, attr)
+
+	obj = getattr(obj, "Preprocessors")
+	obj = getattr(obj,"team_add_info")
+
+	return obj
+
+def get_context(contrib_fname: str, ignore_io_errors: bool, **kwargs):
+	"""
+	Load contrib yaml as base context and add information by context preprocessor for team
+	"""
+	with open(contrib_fname) as f:
+		context = yaml.safe_load(f)
+
+	context["ignore_io_errors"] = ignore_io_errors
+	context.update(kwargs)
+
+	preprocessors = (
+			get_callable(context_prep)
+			for context_prep in context["main"]["context_preprocessors"]
+	)
+
+	for preprocessor in preprocessors:
+		context = preprocessor(context)
+		msg = f"{preprocessor.__name__} is missing the return statement"
+		assert context is not None, msg
+
+	return context
+
+def get_source_files(source_path: str) -> typing.Generator[str, None, None]:
+	for root, dirs, fnames in os.walk(source_path):
+		root = os.path.relpath(root, source_path)
+		for fname in fnames:
+			yield os.path.join(root, fname)
+
+def extend_base_template(content: str, base_template: str) -> str:
+	result = '{% extends "' + base_template + '" %}'
+	result += '{% block body %}'
+	result += content
+	result += '{% endblock %}'
+	return result
+
+def main(source_path: str,
+		 target_path: str,
+		 stored_context: str,
+		 ignore_io_errors: bool) -> int:
+	"""
+	Copy all files in source dir to the target dir
+
+	For ``.md`` and ``.html`` files, render them with context
+	before copying them. ``.md`` files are transformed to HTML.
+	"""
+
+	contrib_fname = os.path.join(source_path, "team.yml")
+	shutil.rmtree(target_path, ignore_errors=True)
+	os.makedirs(target_path, exist_ok=True)
+	sys.stderr.write("Generating context...\n")
+	context = get_context(contrib_fname, ignore_io_errors, stored_context=stored_context)
+	sys.stderr.write("Context generated successfully. \n")
+	templates_path = os.path.join(source_path,
+								  context["main"]["templates_path"])
+	jinja_env = jinja2.Environment(
+					loader=jinja2.FileSystemLoader(templates_path))
+
+	for fname in get_source_files(source_path):
+		if fname in context["main"]["ignore"]:
+			continue
+
+		sys.stderr.write(f"Processing {fname}\n")
+		dirname = os.path.dirname(fname)
+		os.makedirs(os.path.join(target_path, dirname), exist_ok=True)
+
+		extension = os.path.splitext(fname)[-1]
+		if extension in (".html", ".md"):
+				with open(os.path.join(source_path, fname)) as f:
+					content = f.read()
+				if extension == ".md":
+					body = markdown.markdown(
+									content,
+									extensions=context["main"]["markdown_extensions"]
+					)
+					content = extend_base_template(
+									body,
+									context["main"]["base_template"])
```

Comment:
more misaligned code. please only use spaces, no tabs. you should set your code editor to always convert tabs to 4 spaces for any Python code.